logstash-core/lib/logstash/config/config_ast.rb

```diff
@@ -108,6 +108,7 @@ def compile
         # defines @filter_func and @output_func
 
         definitions << "def #{type}_func(event)"
```

Comment:
There is an error here when you run multiples pipeline and the way we are generating the code.
the `def` need to be a `define_singleton_method`

I have a PR to fix it in https://github.com/elastic/logstash/pull/4298